client/src/components/context/auth/authReducer.js

```diff
@@ -37,8 +37,8 @@ export default (state, action) => {
       return {
         ...state,
         token: null,
+        userId: null,
```

Comment:
a good practice for `switch` statements would be to store the return value in a constant, and do a return at the end. 

in addition, while it is not necessary, but good for debugging and maintainability, would be to `break` out of each case even if you're returning something

https://www.w3schools.com/js/js_switch.asp